### ISSUE TITLE
Mapping in child injector

### ISSUE THREAD

**Issue by bullno1** [soundstep/infuse.js#6]
Not sure if this is a bug or it is the intended behavior:

```javascript
var Parent = function() {
    this.mappingInChild = null;
};

var injector = new infuse.Injector();
injector.mapClass("parent", Parent);

var childInjector = injector.createChild();
childInjector.mapValue("mappingInChild", "value");

console.log(childInjector.getValue("parent").mappingInChild);//this will print null
```

I'd expect `mappingInChild` to be `"value"` instead.

**Comment by soundstep** [comment]
This is not an issue.

You are instantiating the parent class, with the main injector:

```
injector.getValue("parent")
```

The ```injector``` does not know anything about a mapping name of ```mappingInChild ``` as you created this mapping name with a child injector, so it can't fulfil its dependencies.

The following will work:

```
console.log(childInjector.getValue("parent").mappingInChild);
```

The rule is: a child injector can have its own mapping names, and will also resolve the one of its parent. A parent injector however, will not be able to resolve the dependencies of its children.

See it as a namespace.

**Comment by bullno1** [comment]
Sorry that was a typo, I meant to do this instead:

```javascript
console.log(childInjector.getValue("parent").mappingInChild);
```

Tested against a3b565e3bf785fe612cde47ec69857cb5eef4d4a and it prints `null`

**Comment by soundstep** [comment]
No worries, I'll have a look tomorrow.

**Comment by soundstep** [comment]
Nicely spotted, the child mapping was not found because if was created from parent mapping. The logic was wrong. I've got a fixed that I will push in the next hour.

**Comment by soundstep** [comment]
Fixed in latest 0.7.4
3 unit tests have been added to cover this.

Demo there:
http://plnkr.co/edit/BFqQGgiqmBokyhRrMW5r?p=preview
